### DISCUSSION TITLE
"Arthur Bench: The Most Robust Way to Evaluate LLMs"

### DISCUSSION THREAD

**Discussion by ianscrivener** [ggml-org/llama.cpp#2826] [Ideas]
_**open-source evaluation tool for comparing LLMs, prompts, and hyperparameters for generative text models.**_

[Arthur AI blog post](https://www.arthur.ai/blog/introducing-arthur-bench)
[https://github.com/arthur-ai/bench](https://github.com/arthur-ai/bench)

@ggerganov, did the Azure Cloud resources ever eventuate?
Perplexity score, Hellaswag scores, Arthur Bench tests... while CI/CD for llama.cpp is ✅done... inference quality and latency benchmarking for llama.cpp would be great to see!

_(with only a MacBook + 4G internet access - personally I'm quite constrained as to what I can do without cloud resources)_

**Comment by ggerganov** [comment]
> did the Azure Cloud resources ever eventuate?

Yes, we currently have 5 nodes (1 of them is V100 GPU, the others are x64 and ARM CPUs):

![image](https://github.com/ggerganov/llama.cpp/assets/1991296/d9b4697a-c4b4-4269-80bd-992d77720886)

I'm fairly happy with it so far. There are lots of things to improve, but for now it does a good job.
More info here: https://github.com/ggerganov/llama.cpp/tree/master/ci

**Reply by ianscrivener** [reply]
Great! Are you doing automated Perplexity and/or Hellaswag bench marking?
